### DISCUSSION TITLE
How Can I Debug: "Error starting server TypeError: Cannot read properties of undefined (reading 'prototype')"

### DISCUSSION THREAD

**Discussion by dylviz** [medusajs/medusa#3959] [Q&A]
hi,

I've spent a few hours trying to figure this out and am not having too much luck. I've read through some of the discussions and looked through issues, but didn't find anything to help.

I am new to backend development and would appreciate some guidance on how I can efficiently debug the following error with starting my server:
```
$ npm run start

> medusa-starter-default@0.0.1 start
> cross-env npm run build && medusa start


> medusa-starter-default@0.0.1 build
> cross-env npm run clean && tsc -p tsconfig.json


> medusa-starter-default@0.0.1 clean
> cross-env ./node_modules/.bin/rimraf dist

[medusa-config] ⚠️ redis_url not found. A fake redis instance will be used.
info:    Using fake Redis
✔ Models initialized – 39ms
✔ Plugin models initialized – 14ms
✔ Strategies initialized – 56ms
⠋ Initializing modules
✔ Modules initialized – 41ms
✔ Database initialized – 1303ms
✔ Repositories initialized – 42ms
✔ Services initialized – 88ms
✔ Express intialized – 9ms
⠋ Initializing plugins
error:   Error starting server
TypeError: Cannot read properties of undefined (reading 'prototype')
    at registerPaymentServiceFromClass (...MedusaBackend/node_modules/@medusajs/medusa/dist/loaders/helpers/plugins.js:8:51)
    at ...MedusaBackend/node_modules/@medusajs/medusa/dist/loaders/plugins.js:363:87
    at step (...MedusaBackend/node_modules/@medusajs/medusa/dist/loaders/plugins.js:33:23)
    at Object.next (...MedusaBackend/node_modules/@medusajs/medusa/dist/loaders/plugins.js:14:53)
    at ...MedusaBackend/node_modules/@medusajs/medusa/dist/loaders/plugins.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (...MedusaBackend/node_modules/@medusajs/medusa/dist/loaders/plugins.js:4:12)
    at ...MedusaBackend/node_modules/@medusajs/medusa/dist/loaders/plugins.js:354:87
    at Array.map (<anonymous>)
    at ...MedusaBackend/node_modules/@medusajs/medusa/dist/loaders/plugins.js:354:60

```

I followed this [tutorial](https://medusajs.com/blog/extending-medusa-usecase-marketplace/#create-migration-for-the-user-entity) and everything was working. Then continued onto the "next step" suggestions of extending the Order entity and other things following the docs and the way it was shown in the tutorial, nothing crazy.

I did switch to using a potgresql database on ubuntu (that was working fine as well) but I don't think that's a problem because the same error happens when I go back to using the default sqlite.

I'm using the default plugin config:
```
const plugins = [
	`medusa-fulfillment-manual`,
	`medusa-payment-manual`,
	// To enable the admin plugin, uncomment the following lines and run `yarn add @medusajs/admin`
	// {
	// 	resolve: "@medusajs/admin",
	// 	/** @type {import('@medusajs/admin').PluginOptions} */
	// 	options: {
	// 		autoRebuild: true,
	// 	},
	// },
];
```

Any tips would be appreciated!
Thanks

**Comment by gaurav0909-max** [comment]
same happened with me 
any solution ?

**Comment by oliwerhelsen** [comment]
One thing can be that you forgot to export default any service. Export of a class is not enough.

**Comment by supermarios77** [comment]
Same, Any answer yet?

**Reply by oliwerhelsen** [reply]
Can you provide some more info. Maybe post the OrderService class?

**Reply by supermarios77** [reply]
Its working now i created an outdated project so now i just deleted it and created a new updated one

**Comment by srindom** [comment]
@dylviz did you find a solution to this? 

@oliwerhelsen's answer is a good first check :)

**Comment by chumaumenze** [comment]
Ensure that JS files in your service directory contains and exports a default service. An unexpected JS file can trigger this error.

cc: @supermarios77 @gaurav0909-max

**Comment by Kushal-starr** [comment]
Pls check you should have .env file

**Comment by miguel-martinez-6661** [comment]
Make sure you don't have any `blank` file or without `export default` in the `backend/services` directory

**Comment by mattmungai** [comment]
i had a problem with service undefined. To fix it I had to explicitly define it in my index.js file not pass in const HELLO_MODULE

export const HELLO_MODULE = "helloModuleService";

export default Module("helloModuleService", {
  service: HelloModuleService, // This is the service key
});
